### ISSUE TITLE
why new defined op always runs on PS host, not in worker host?

### ISSUE THREAD

**Issue by yuanwujun** [tensorflow/tensorflow#19688]
i have refactored word2vec code（from models/tutorials/embedding/）, my problem is new defined OP "train_op = word2vec.neg_train_word2vec "  runs on PS,  I don't know where I went wrong.  my code :           
with tf.device(tf.train.replica_device_setter(worker_device="/job:worker/task:%d" % FLAGS.task_index,
                                                      cluster=cluster, ps_strategy=ps_strategy, merge_devices=False)):

            # Declare all variables we need.
            embedding = tf.get_variable(name="word_embedding", shape = [71291,FLAGS.embedding_size],
                        initializer = tf.random_uniform_initializer(-0.5 / FLAGS.embedding_size, 0.5 / FLAGS.embedding_size))

            weights =   tf.get_variable(name="word_weights",shape = [71291, FLAGS.embedding_size],
                        initializer = tf.random_normal_initializer() )

            global_step = tf.Variable(0, name="global_step", trainable=False)

            # The training data
            ( words,
              counts,
              words_per_epoch,
              current_epoch,
              total_words_processed,examples,
              labels) = word2vec.skipgram_word2vec(filename=FLAGS.train_data,
                                           batch_size=FLAGS.batch_size,
                                           window_size=FLAGS.window_size,
                                           min_count=FLAGS.min_count,
                                           subsample=FLAGS.subsample)

            # Linear learning rate decay.
            words_to_train = tf.cast(words_per_epoch * FLAGS.epochs_to_train, tf.float32)
            lr = FLAGS.learning_rate * tf.maximum(
                        0.0001,
                        1.0 - tf.cast(total_words_processed, tf.float32) / words_to_train)

            train_op = word2vec.neg_train_word2vec(embedding,
                                          weights,
                                          examples,
                                          labels,
                                          lr,
                                          #vocab_count=counts.tolist(),
                                           num_negative_samples=FLAGS.num_neg_samples)

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by yuanwujun** [comment]
OS Platform and Distribution: linux   centos7.1
TensorFlow installed from: pip install tensorflow
Tensorflow version: 1.7.0 cpu-version
CUDA: 9.0
cuDNN:7.2
GPU:P40 memory:22G

**Comment by angersson** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!
